### DISCUSSION TITLE
AttributeError: 'Attribute' object has no attribute 'id' when using "Unicorn.call("component","method","parameter")

### DISCUSSION THREAD

**Discussion by Jocelyn-AGORO** [django-commons/django-unicorn#337] [General]
Error when using Unicorn.call() on a page to  call backend method on component.

**Comment by adamghill** [comment]
Hi, can you give me more some more information about this error? Best way would be to fork this repo and replicate it using https://github.com/adamghill/django-unicorn/blob/master/example/www/templates/www/index.html#L8 and https://github.com/adamghill/django-unicorn/blob/master/example/unicorn/components/hello_world.py#L11 or a unit test that demonstrates the issue. Second best would be a small piece of code in this discussion that replicates the problem.

Thanks!

**Reply by Jocelyn-AGORO** [reply]
Thank you for your answer, I'm going to explain. I have two components on the same page 'filter' and 'articles'  and a menu in my filter components listing different categories. I want  that when user click on a specific categorie articles in 'articles' component get updated.

In filter component :
```
{% for categorie in categories %}
                    <li>
                        <a u:click="Unicorn.call('articles','filter',{{ categorie.id }})" href="#">
                            {{categorie.title}}<span class="category-counter"> ({{categorie.num_articles}})</span>
                        </a>
                    </li>
{% endfor %}
```
In articles:
```
class ArticlesView(UnicornView):
    def filter(self, id):
        self.articles = Article.objects.filter(categorie_id=id)
```

template:
```
{% extends 'layout.html' %}
{% load static %}
{% load tedy_extras %}
{% load unicorn %}
{% block title %}
       filter 
{% endblock %}

{% block layout %}
    <div class="main-container section-padding">
    <div class="container">
        <div class="row mt-5">
            {% unicorn 'filter' %}
            {% unicorn 'articles' %}
        </div>
    </div>
</div>
{% endblock %}
```
